recipes/generate.py

```diff
@@ -54,14 +54,22 @@ def _setup_model(
         model_cfg: DictConfig,
         model_state_dict: Dict[str, Any],
     ) -> nn.Module:
-        with utils.set_default_dtype(self._dtype), self._device:
+        with torch.device("meta"):
             model = config.instantiate(model_cfg)
 
+        model.load_state_dict(model_state_dict, assign=True)
```

Comment:
for quantized models we'd need to load after we do quantization I think

Reply:
We should consider doing this https://docs-preview.pytorch.org/pytorch/tutorials/2824/recipes/recipes/swap_tensors.html?highlight=swap_tensor once we can use 2.3+ in AO/here.